File: test_c/test_fibit.c
#include <stdio.h>



long long fibit(int n, int *opC) {
    if (n <= 2) { return 1; }
    long long n_1 = 1;
    long long n_2 = 1;
    for (int i = 3; i <= n; i++) {
        long long current = n_1 + n_2;
        (*opC)++;
        n_1 = n_2;
        n_2 = current;
    }
    return n_2;
}

int main() {
    int opC = 0;
    int passed = 0;
    if (fibit(1, &opC) == 1)                        { printf("test 1 passed\n"); passed++; }
    if (fibit(2, &opC) == 1)                        { printf("test 2 passed\n"); passed++; }
    if (fibit(7, &opC) == 13)                       { printf("test 3 passed\n"); passed++; }
    if (fibit(10, &opC) == 55)                      { printf("test 4 passed\n"); passed++; }
    if (fibit(20, &opC) == 6765)                    { printf("test 5 passed\n"); passed++; }
    if (fibit(93, &opC) == 12200160415121876738ULL) { printf("test 6 passed\n"); passed++; }
    printf("%d/6 tests passed.\n", passed);
    return 0;
}

File: test_rs/test_fibit.rs
fn fibit(n: u64, op_c: &mut u64) -> u64 {
    if n <= 2 { return 1; }
    let mut n_1: u64 = 1;
    let mut n_2: u64 = 1;
    for _i in 3..=n {
        let current = n_1 + n_2;
        *op_c += 1;
        n_1 = n_2;
        n_2 = current;
    }
    n_2
}

fn main() {
    let mut op_c: u64 = 0;
    let mut passed = 0;
    if fibit(1, &mut op_c) == 1                      { println!("test 1 passed"); passed += 1; }
    if fibit(2, &mut op_c) == 1                      { println!("test 2 passed"); passed += 1; }
    if fibit(7, &mut op_c) == 13                     { println!("test 3 passed"); passed += 1; }
    if fibit(10, &mut op_c) == 55                    { println!("test 4 passed"); passed += 1; }
    if fibit(20, &mut op_c) == 6765                  { println!("test 5 passed"); passed += 1; }
    if fibit(93, &mut op_c) == 12200160415121876738  { println!("test 6 passed"); passed += 1; }
    println!("{}/6 tests passed.", passed);
}
